File: FootballExplorer/app/src/main/java/fr/a42consulting/footballexplorer/Model/TeamsSearchModel.java
package fr.a42consulting.footballexplorer.Model;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class TeamsSearchModel {
    public static TeamsSearchModel fromJson(JSONObject jsonObject) {
        TeamsSearchModel league = new TeamsSearchModel();
        try {
            JSONArray jsonArray = jsonObject.getJSONArray("teams");

            JSONObject teamJson;
            ArrayList<TeamsSearchTeam> teams = new ArrayList<TeamsSearchTeam>(jsonArray.length());
            // Process each result in json array, decode and convert to team object
            for (int i=0; i < jsonArray.length(); i++) {
                try {
                    teamJson = jsonArray.getJSONObject(i);
                } catch (Exception e) {
                    e.printStackTrace();
                    continue;
                }

                TeamsSearchTeam team = TeamsSearchTeam.fromJson(teamJson);
                if (team != null) {
                    teams.add(team);
                }
            }

            league.teams = teams;

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return league;
    }
    private ArrayList<TeamsSearchTeam> teams = new ArrayList<>();

    public ArrayList<TeamsSearchTeam> getTeams() {
        return teams;
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/model/TeamsSearchModel.kt
package fr.a42consulting.footballexplorer.model

import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

class TeamsSearchModel {
    var teams = ArrayList<TeamsSearchTeam>()
        private set

    companion object {
        fun fromJson(jsonObject: JSONObject): TeamsSearchModel {
            val league = TeamsSearchModel()
            try {
                val jsonArray = jsonObject.getJSONArray("teams")

                var teamJson: JSONObject
                val teams = ArrayList<TeamsSearchTeam>(jsonArray.length())
                // Process each result in json array, decode and convert to team object
                for (i in 0 until jsonArray.length()) {
                    try {
                        teamJson = jsonArray.getJSONObject(i)
                    } catch (e: Exception) {
                        e.printStackTrace()
                        continue
                    }

                    val team = TeamsSearchTeam.fromJson(teamJson)
                    if (team != null) {
                        teams.add(team)
                    }
                }

                league.teams = teams

            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return league
        }
    }
}
